src/catalog/metastore.rs

```diff
@@ -178,3 +213,21 @@ impl Metastore {
         })
     }
 }
+
+impl ObjectStoreFactory for Metastore {
+    fn parse_url_opts(
+        &self,
+        url: &Url,
+        _options: &StorageOptions,
+    ) -> DeltaResult<(ObjectStoreRef, Path)> {
+        if self.default_store.root_uri.scheme() == url.scheme() {
+            Ok((
+                Arc::new(self.default_store.as_ref().clone()),
+                Path::from("/"),
+            ))
+        } else {
+            // TODO: this won't work if the default store has the same scheme as the queried one
```

Comment:
This is benign, this pathway doesn't even take place, since we instantiate all the tables with object stores (and not url + object store options).

The only reason we need to register these in the first place is so that delta-rs knows all allowed schemes. Will return an Err here explicitly to clarify that.

Reply:
nice